src/fixup/fixup.cpp

```diff
@@ -113,38 +150,135 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
     params.Add("ceiling", Ceilings());
   }
 
+  if (enable_rad_floors) {
+    const std::string floor_type = pin->GetString("fixup", "rad_floor_type");
+    if (floor_type == "ConstantJ") {
+      Real J0 = pin->GetOrAddReal("fixup", "J0_floor", 0.0);
+      params.Add("rad_floor", RadiationFloors(constant_j_floor_tag, J0));
+    } else if (floor_type == "ExpX1J") {
+      Real J0 = pin->GetOrAddReal("fixup", "J0_floor", 0.0);
+      Real Jp = pin->GetOrAddReal("fixup", "J_exp_floor", -2.0);
+      params.Add("rad_floor", RadiationFloors(exp_x1_j_floor_tag, J0, Jp));
+    } else {
+      PARTHENON_FAIL("invalid <fixup>/rad_floor_type input");
+    }
+  } else {
+    params.Add("rad_floor", RadiationFloors());
+  }
+
+  if (enable_rad_ceilings) {
+    const std::string radiation_ceiling_type =
+        pin->GetOrAddString("fixup", "radiation_ceiling_type", "ConstantXi0");
+    if (radiation_ceiling_type == "ConstantXi0") {
+      Real xi0 = pin->GetOrAddReal("fixup", "xi0_ceiling", 0.99);
+      params.Add("rad_ceiling",
+                 RadiationCeilings(constant_xi0_radiation_ceiling_tag, xi0));
+    }
+  } else {
+    params.Add("rad_ceiling", RadiationCeilings());
+  }
+
   if (enable_mhd_floors) {
-    Real bsqorho_max = pin->GetOrAddReal("fixup", "bsqorho_max", 50.);
-    Real bsqou_max = pin->GetOrAddReal("fixup", "bsqou_max", 2500.);
-    Real uorho_max = pin->GetOrAddReal("fixup", "uorho_max", 50.);
-    params.Add("bsqorho_max", bsqorho_max);
-    params.Add("bsqou_max", bsqou_max);
-    params.Add("uorho_max", uorho_max);
+    const std::string mhd_ceiling_type =
+        pin->GetOrAddString("fixup", "mhd_ceiling_type", "ConstantBsqRat");
+    if (mhd_ceiling_type == "ConstantBsqRat") {
+      Real bsqorho0 = pin->GetOrAddReal("fixup", "bsqorho0_ceiling", 50.);
+      Real bsqou0 = pin->GetOrAddReal("fixup", "bsqou0_ceiling", 2500.);
+      params.Add("mhd_ceiling",
+                 MHDCeilings(constant_bsq_rat_ceiling_tag, bsqorho0, bsqou0));
+    } else {
+      PARTHENON_FAIL("invalid <fixup>/mhd_ceiling_type input");
+    }
+  } else {
+    params.Add("mhd_ceiling", MHDCeilings());
   }
 
+  FAILURE_STRATEGY fluid_c2p_failure_strategy;
+  std::string fluid_c2p_failure_strategy_str =
+      pin->GetOrAddString("fixup", "fluid_c2p_failure_strategy", "interpolate");
+  if (fluid_c2p_failure_strategy_str == "interpolate") {
+    fluid_c2p_failure_strategy = FAILURE_STRATEGY::interpolate;
+  } else if (fluid_c2p_failure_strategy_str == "interpolate_previous") {
+    fluid_c2p_failure_strategy = FAILURE_STRATEGY::interpolate_previous;
+  } else if (fluid_c2p_failure_strategy_str == "neighbor_minimum") {
+    fluid_c2p_failure_strategy = FAILURE_STRATEGY::neighbor_minimum;
+  } else if (fluid_c2p_failure_strategy_str == "floors") {
+    fluid_c2p_failure_strategy = FAILURE_STRATEGY::floors;
+  }
+  params.Add("fluid_c2p_failure_strategy", fluid_c2p_failure_strategy);
```

Comment:
`interpolate` averages the updated primitives from good neighbors. `interpolate_previous` uses the primitives from the previous substep for interpolation. `neighbor_minimum` uses updated good primitives but takes the minimum of each primitive over the good neighbors. `floors` sets everything to floors. 

Since this ended up not being necessary I should probably just remove everything except `interpolate` (which falls back to `floors` in the case of no good neighbors) and `floors`.

Reply:
I would be in favor of removing the unused versions :+1:

Reply:
Removed :+1: